packages/e2e-test-utils-playwright/src/editor/switch-block-inspector-tab.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Internal dependencies
+ */
+import type { Editor } from './index';
+import { expect } from '../test';
+
+/**
+ * Clicks on the block inspector tab button with the supplied label and waits
+ * for the tab switch.
+ *
+ * @param {Editor} this
+ * @param {string} label
+ */
+export async function switchBlockInspectorTab( this: Editor, label: string ) {
+	const inspectorTabButton = this.page.locator(
+		`role=region[name="Editor settings"i] >> role=tab[name="${ label }"i]`
+	);
+	const id = await inspectorTabButton.getAttribute( 'id' );
+
+	await inspectorTabButton.click();
+	await expect(
+		this.page.locator(
+			`role=region[name="Editor settings"i] >> div[role="tabpanel"][aria-labelledby="${ id }"]`
+		)
+	).toBeVisible();
```

Comment:
Since we're using `aria-labelledby`, we can reuse the same `label` for the tab to avoid relying on implementation details. This util can be rewritten into:

```ts
const editorSettings = this.page.locator( 'role=region[name="Editor settings"i]' );

await editorSettings.locator( `role=tab[name="${ label }"i]` ).click();

await expect( editorSettings.locator( `role=tab[name="${ label }"i]` ) ).toBeVisible();
```

But then again, as you can see the util is now fairly straightforward and be inlined in most cases. We can just do this in playwright test:

```js
page.click( `role=region[name="Editor settings"i] >> role=tab[name="${ label }"i]` );
```

We don't really need the last `expect` assertion if we're inlining it. Because most actions below it will wait for the actionability implicitly anyway.

Reply:
Thanks for the suggested improvements, I've implemented them within https://github.com/WordPress/gutenberg/pull/47097